agents/ui.py

```diff
@@ -271,7 +271,12 @@ def __init__(self, text, func, model):
         self.model = model
         self.func = func
         self.timer = QtCore.QTimer()
-        self.timer.timeout.connect(lambda: self.func(model))
+
+        def call_func():
+            if not model.is_paused():
+                self.func(model)
+
+        self.timer.timeout.connect(call_func)
```

Comment:
I princippet ville det nok være pænere hvis den ikke stod og lavede en masse events hvert sekund, som aldrig bliver kørt, så længe den er paused, men jeg tror vi på et tidspunkt skal rydde op i hele logikken omkring opdateringer af de forskellige komponenter, så der ikke er tusind forskellige timers, men der er et form for "main loop". Dvs. det gør måske ikke så meget at det er ineffektivt lige nu - vi optimerer det senere.